doc/source/advancedexamples.rst

```diff
@@ -445,6 +466,51 @@ be written using :class:`qibo.base.gates.VariationalLayer` as follows:
         circuit.add(gates.CZ(0, nqubits - 1))
     circuit.add((gates.RY(i, theta) for i in range(nqubits)))
 
+.. _vqc-example:
+
+How to write a custom VQC optimization?
+---------------------------------------
+
+Similarly to the VQE, a custom implementation of a VQC model can be achieved by
+defining a custom loss function and calling the :meth:`qibo.optimizers.optimize`
+method.
+
+Here is a simple example using a custom loss function:
+
+.. code-block:: python
+
+    import numpy as np
+    from qibo import models, gates
+    from qibo.optimizers import optimize
+
+    # custom loss function which returns numpy object
+    def myloss(parameters, circuit, target):
+        circuit.set_parameters(parameters)
+        return np.square(np.sum(circuit()-target))
```

Comment:
Note that this will return a complex number for the loss because `circuit()` will be complex. A more appropriate loss would be 
```Python
return np.sum(np.square(np.abs(circuit() - target)))
```
Even better we could use the fidelity which is the common measure of distance between quantum states:
```Python
return 1 - np.abs(np.conj(target).dot(circuit()))
```